### ISSUE TITLE
How about supporting ARM64.

### ISSUE THREAD

**Issue by pcfulife** [dotnet/wpf#1817]
Windows 10 supports ARM64 processors. And, Windows 10 on ARM64 can run x86 applications. But, performance is significantly low because of emulation. And, it cannot run native AMD64 applications. So, some developers recompile their applications for better performance.

Win32 API supports ARM processors. (I saw native ARM64 applications which is not an UWP app) But, WPF does not support ARM64 but only x86/x64. If WPF supports ARM64, many WPF applications can be run efficiently with ARM processors.

**Comment by dbeavon** [comment]
Yes!  Can't wait until my wpf apps are running natively on arm64 for 2 days on battery... W/ snapdragon 8cx and a 5g connection to the internet. 

I might just have to stop buying expensive smartphones and start putting that money towards my arm64 laptop instead.  It's a much better value.

**Comment by BigBadBleuCheese** [comment]
The next version of my WPF app will be moving from .NET Framework to .NET Core. I would absolutely love to make an ARM64 binary available to customers using the new Surface Pro X.

**Comment by shibayan** [comment]
I purchased and verified Surface Pro X, but x86 emulation on ARM64 was more overhead than I thought. (The performance seems to have dropped by about 2/3.)

ARM64 support for .NET Core and WPF is very important because it is likely that you will not be able to experience the customer's expectations.

**Comment by BigBadBleuCheese** [comment]
I totally agree, @shibayan. I'm a fairly experienced WPF developer, I picked up my Pro X today, and I stand ready to be a dutiful guinea pig of this team. ;)

**Comment by soapdog** [comment]
What about support for Windows on ARM?

**Comment by JtheOne** [comment]
Who do we talk to make WPF Windows ARM64 support a priority 🤔

**Comment by IAIKSysadminTeam** [comment]
+1 @JtheOne. Would be important for many applications to support native arm64 apps. Emulation is quite a pain at the moment!

**Comment by soapdog** [comment]
Just as an example, Serif who makes the Affinity suite of products (Affinity Designer, Photo, and Publisher) answered me on Twitter that they can't port their apps to the Surface Pro X because WPF is not there.

**Comment by MikhailTNY** [comment]
MS updated their roadmap for WPF 5/ARM: https://github.com/dotnet/wpf/blob/master/roadmap.md

ETA for ARM is 21H1.

**Comment by predavid** [comment]
Stay tuned for an announcement on WPF ARM64 support early next month. Details will be published here on GitHub.

**Comment by predavid** [comment]
dotnet/wpf#4048  is a recent pull request pointing at a few other ARM64 related issues. We are aiming to have WPF ARM64 support in the upcoming .NET6 Preview1.

**Comment by driver1998** [comment]
Looks like WPF is already working in .NET 6 nightly:
https://twitter.com/sinclairinat0r/status/1354987172324122624?s=19

Image from the tweet:
![image](https://user-images.githubusercontent.com/22699485/106228581-a0bdba80-6226-11eb-8a1c-079911df86af.png)

**Comment by ryalanms** [comment]
The link isn't up on https://github.com/dotnet/installer yet, but you can get the build with ARM64 WPF support here:

https://aka.ms/dotnet/net6/dev/Sdk/dotnet-sdk-win-arm64.exe (version 6.0.100-preview.1.21072.23 or later)

You can also target ARM64 from an x64/x86 machine.

**Comment by kpreisser** [comment]
Thank you for adding ARM64 support to WPF! I was able to run `CefSharp.MinimalExample.Wpf.netcore` (Chromium embedded in WPF application using offscreen rendering) successfully as ARM64 with .NET SDK `6.0.100-preview.1.21078.9` (see https://github.com/cefsharp/CefSharp/pull/3359) :
![grafik](https://user-images.githubusercontent.com/13289184/106354650-a8976080-62f3-11eb-923c-f56e26ccd470.png)

Is it still planned to include WPF ARM64 support in an upcoming .NET 5.0 servicing update? (WPF is the last missing piece before we can publish the software developed at my workplace for Windows 10 ARM64, and we hope that we don't have to wait for the .NET 6.0 release.)
Thanks!

**Comment by predavid** [comment]
Thank you @kpreisser for trying out the build. 

A .NET 5.0 servicing update is under discussion/planning. We are heads-down on getting some new devs recruited and onboarded which will increase the pace and the scope of our future investments.Please stay tuned for an update in March which will include a projected timeline.  Could you please share more information is possible on the nature of your software, how many employees will be using it ..etc?

**Comment by vatsan-madhavan** [comment]
Congrats @dotnet/wpf-developers on getting ARM64 support out - nice job! 🚀

**Comment by kpreisser** [comment]
> A .NET 5.0 servicing update is under discussion/planning.

Ok, thank you! Actually, we don't have a strong requirement to publish a Windows ARM64 build of our app soon (most customers run it on Windows x64, or (the CLI variant) on Linux ARM/x64), but it would be a nice-to-have if we could publish it in the next few months for Win ARM64, so that it can run at native speed on devices such as the Surface Pro X.

**Comment by predavid** [comment]
Thank you @kpreisser . We are taking into consideration your requirement for .NET5 ARM64 support for the latter half of the year. We will keep fine tuning our roadmap to reflect newer/upcoming investments. Please stay tuned. 

In the meantime , do open Issues for any new bugs you encounter with the .NET 6 ARM64 support.

**Comment by Gerdya** [comment]
Hi @predavid. Did i miss something because the WPF support is on the roadmap for .NET5 for 21H1 at least according to 

https://github.com/dotnet/wpf/blob/master/roadmap.md

Now you suddenly start talking about latter half? Not sure if you are aware that the availability of many applications for ARM64 are blocked by the availability of WPF.

**Comment by predavid** [comment]
Hi @Gerdya, we are considering feedback from each of our customers who express interest in ARM64 support to educate and refine the timeline for our future investments based on new dev resources coming on-board. So, while @Kpreisser might need it only in the later half of the year, based on your inputs, I believe you would need it sooner? 
Have you had a chance to try out the WPF ARM64 support on .NET 6 Preview 1? We'd love to hear of any bugs/issues you may encounter. 

Also, if possible we'd love to know of the type of your app, how many apps (if more than 1), apprx how users per app, that would be running on ARM64.

**Comment by Alovchin91** [comment]
Hi @predavid, do you have any updates regarding the matter? Many apps are still blocked by the lack of WPF support on ARM64.

**Comment by fabiant3** [comment]
@Alovchin91 - see above response from Premalini "Have you had a chance to try out the WPF ARM64 support on .NET 6 Preview 1? We'd love to hear of any bugs/issues you may encounter." This shipped with NET6 Preview1.

**Comment by Alovchin91** [comment]
Hi @fabiant3, thanks for the response! I've seen .NET 6 Previews indeed. But my question was about WPF support on ARM64 that was originally promised for .NET 5. Do you have any news on .NET 5?

**Comment by fabiant3** [comment]
We are considering servicing NET5 with WPF ARM64 support. We will do all of our validation on NET6 and then port to NET5. I do not have a timeline just yet, but this is committed and expected to ship this year.

**Comment by Alovchin91** [comment]
Ah, clear about validation on .NET 6 👍 Thanks for the clarification! 🙂

**Comment by predavid** [comment]
@Alovchin91 - any validation on .NET6 that you can give us feedback on (similar to kpreisser above) will help us to quicken getting to Quality on .NET 6 and then commencing on servicing .NET5. The more the Community validates ARM64 .NET6, the better.  Thanks in advance!

**Comment by Samuel12321** [comment]
@predavid unfortunately there is a bug #4336 preventing apps that support multiple languages building wpf .net 6 for arm64

**Comment by Samuel12321** [comment]
Turns out visual studio's ALINK (al.exe) doesn't support building for arm64. A workaround is available.
https://developercommunity.visualstudio.com/t/alink-alexe-does-not-support-arm64-windows/1325460

**Comment by Samuel12321** [comment]
@fabiant3 what happened to ArRM64 wpf support for .net 5 in 21H1
https://github.com/dotnet/wpf/blob/main/roadmap.md

**Comment by Alovchin91** [comment]
@Samuel12321  It has arrived: https://devblogs.microsoft.com/dotnet/net-july-2021/#windows-arm64

**Comment by Samuel12321** [comment]
@Alovchin91 thanks for pointing that out , I didn't see that post. 
The roadmap probably should be updated.
Good to see this finally added.

**Comment by wmjb** [comment]
why no ARM 32 support? thousands of devices out there could do with a fresh bit of software. ignoring this architecture allows planned obsolesce of hardware. thousands of windows rt devices could happily run windows 10 (and already are with leaked preview) and with a new browser they could be useful devices. are we supposed to just put them in the bin?
